### DISCUSSION TITLE
Map not loading?

### DISCUSSION THREAD

**Discussion by MikeBishop** [teslamate-org/teslamate#1394] [Q&A] [answered]
I recently moved my Teslamate container from one host to another -- did a backup on one host, moved the docker-compose over to the other host, started there and restored the backup.  Everything looks great, except for one issue:

On the main page of Teslamate, where there used to be a map showing the car's current location, I'm just getting a spinner.  Is this a more general issue, or is there something I should be doing to debug it locally?

**Comment by Dulanic** [comment]
I have had this same problem for a LONG time, and could never figure out the problem myself. Did you ever get anywhere with this?
<img width="1685" height="979" alt="image" src="https://github.com/user-attachments/assets/8d9ea444-985d-4021-9c85-3c5c8cef810d" />

<img width="1117" height="1308" alt="image" src="https://github.com/user-attachments/assets/45a51384-5d0d-4e3d-9eb4-c4385af65623" />

**Comment by Dulanic** [comment, marked as answer]
Fixed it. Updated CHECK_ORIGIN=https://teslamate.dulanic.com,http://192.168.2.155:4000 to include my server IP. I don't think we explain this anywhere.

**Reply by JakobLichterfeld** [reply]
We hint users about it (for example here: https://github.com/teslamate-org/teslamate/issues/4825#issuecomment-3074698740), but this is just a reverse proxy issue, not a TeslaMate issue. And the docs about this env variable are clear as well: https://docs.teslamate.org/docs/configuration/environment_variables

And the log output is quite helpful imo

**Comment by SemoTech** [comment]
If anyone is using a secure Caddy V2 setup, as per [my post](https://github.com/teslamate-org/teslamate/issues/4825#issuecomment-3088638272), I can confirm that the missing tile images is a Caddy secure proxy issue. 

FYI @JakobLichterfeld thanks for your latest hint. Unfortunately I was unable to reply on my thread since it was locked, but the  CHECK_ORIGIN parameter was not the issue when accessing from teslamate.[my-domain].com.  If you dont mind you can post the solution on my thread as well to help anyone else using Caddy and enhanced security...

Now, my console logs showed: `The resource at “https://tile.openstreetmap.org/17/38593/49239.png” was blocked due to its Cross-Origin-Resource-Policy header (or lack thereof).`

This occurs since the OSM’s tile server (tile.openstreetmap.org) returns a Cross-Origin-Resource-Policy: same-origin header, which prevents the Teslamate home page map tiles from being loaded in `<img>` tags on a page served from my domain (teslamate.[my-domain].com) when it/TeslaMate is served through the Caddy reverse proxy over HTTPS with Cross-Origin security enabled.  

The browser blocks these requests because teslamate.[my-domain].com is a different origin than tile.openstreetmap.org, and OSM does not include an Access-Control-Allow-Origin header to allow cross-origin embedding. The CHECK_ORIGIN parameter in TeslaMate’s configuration is not impacting or resolving the issue in this case, because it primarily controls CORS for TeslaMate’s API endpoints, not the external tile server requests.

So this **blocking** in my case was due to the enhanced `Cross-Origin-Resource-Policy "same-origin"` global security header I had present in my Caddy V2 configuration file, and thus when TeslaMate’s frontend  tries to load tiles from https://tile.openstreetmap.org/{z}/{x}/{y}.png, the browser sees the same-origin CORP policy on the parent page (teslamate.[my-domain].com) and blocks the tiles because they come from a different origin (tile.openstreetmap.org) and also have a same-origin CORP header.

The combination of my server's same-origin policy and OSM’s own same-origin policy creates a double restriction, preventing the tiles from being embedded in `<img>` tags!

The solution for me, since I restrict access to teslamate, and allow it only on the LAN, was to relax CORP for TeslaMate security in it's local configuration in Caddy V2 by adding `header_down Cross-Origin-Resource-Policy cross-origin` and `header_down Cross-Origin-Embedder-Policy "credentialless"` to the `reverse_proxy teslamate:4000` in the Caddyfile config which for my setup looks as follows:

<img width="489" height="372" alt="image" src="https://github.com/user-attachments/assets/0d30d6fb-a1f4-47ee-90e7-57e796b74d3c" />

Hope this helps others that use Caddy...
